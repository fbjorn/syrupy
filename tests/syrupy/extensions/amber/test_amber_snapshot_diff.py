def test_snapshot_diff(snapshot):
    my_dict = {
        "field_0": True,
        "field_1": "no_value",
        "nested": {
            "field_0": 1,
        },
    }
    assert my_dict == snapshot
    my_dict["field_1"] = "yes_value"
    assert my_dict == snapshot(diff=0)
    my_dict["nested"]["field_0"] = 2
    assert my_dict == snapshot(diff=0)


def test_snapshot_diff_id(snapshot):
    my_dict = {
        "field_0": True,
        "field_1": "no_value",
        "field_2": 0,
        "field_3": None,
        "field_4": 1,
        "field_5": False,
        "field_6": (True, "hey", 2, None),
        "field_7": {False, "no", 0, None},
    }
    my_dict["nested"] = dict(my_dict)
    assert my_dict == snapshot(index="large snapshot")
    my_dict["field_1"] = "yes_value"
    assert my_dict == snapshot(diff="large snapshot")
    my_dict["nested"]["field_0"] = 2
    assert my_dict == snapshot(index="case3", diff="large snapshot")
