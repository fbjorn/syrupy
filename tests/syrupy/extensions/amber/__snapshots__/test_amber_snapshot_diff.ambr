# name: test_snapshot_diff
  dict({
    'field_0': True,
    'field_1': 'no_value',
    'nested': dict({
      'field_0': 1,
    }),
  })
# ---
# name: test_snapshot_diff.1
      ...
  -   'field_1': 'no_value',
  +   'field_1': 'yes_value',
      ...
# ---
# name: test_snapshot_diff.2
      ...
  -   'field_1': 'no_value',
  +   'field_1': 'yes_value',
      ...
  -     'field_0': 1,
  +     'field_0': 2,
    ...
# ---
# name: test_snapshot_diff_id.'case3'
      ...
  -   'field_1': 'no_value',
  +   'field_1': 'yes_value',
        ...
  -     'field_0': True,
  +     'field_0': 2,
          ...
# ---
# name: test_snapshot_diff_id.'large snapshot'
  dict({
    'field_0': True,
    'field_1': 'no_value',
    'field_2': 0,
    'field_3': None,
    'field_4': 1,
    'field_5': False,
    'field_6': tuple(
      True,
      'hey',
      2,
      None,
    ),
    'field_7': set({
      'no',
      False,
      None,
    }),
    'nested': dict({
      'field_0': True,
      'field_1': 'no_value',
      'field_2': 0,
      'field_3': None,
      'field_4': 1,
      'field_5': False,
      'field_6': tuple(
        True,
        'hey',
        2,
        None,
      ),
      'field_7': set({
        'no',
        False,
        None,
      }),
    }),
  })
# ---
# name: test_snapshot_diff_id.1
      ...
  -   'field_1': 'no_value',
  +   'field_1': 'yes_value',
        ...
# ---
